src/DalService.ts

```diff
@@ -0,0 +1,26 @@
+import { AxiosResponse } from 'axios';
+import GenerateApiError from './utils/GenerateApiError';
+
+export default class DalService {
```

Comment:
Make this abstract. Nobody should ever be able to instantiate this class on it's own.

Reply:
Done